File: ControlBoard/ControlBoard.Domain/Dto/StationDto.cs
using ControlBoard.DB.Entities.Enums;

namespace ControlBoard.Domain.Dto
{
    public class StationDto
    {
        public int Id { get; set; }
        public int ChartElementId { get; set; }
        public string? Name { get; set; }
        public int AreaId { get; set; }
        public string? AreaName { get; set; }
        public ProductTypes? ProductType { get; set; }
    }
}

File: mes-ui/src/app/Entities/StationDto.ts
import {ProductTypes} from '../enums/ProductTypes';
import {ProcessStateDto} from './ProcessStateDto';

export interface StationDto {
  id: number;
  chartElementId: number;
  name: string;
  areaId: number;
  areaName?: string;
  productType?: ProductTypes;
  processStates?: ProcessStateDto[];
}
